client/app/components/admin/CeleryStatus.jsx

```diff
@@ -0,0 +1,220 @@
+    queues[task.queue][task.state] += 1;
+
+    if (task.enqueue_time) {
+      task.enqueue_time = moment(task.enqueue_time * 1000.0);
+    }
+    if (task.start_time) {
+      task.start_time = moment(task.start_time * 1000.0);
+    }
+
+    counters[task.state] += 1;
+
+    if (task.task_name === 'redash.tasks.execute_query') {
+      queries.push(task);
+    } else {
+      otherTasks.push(task);
+    }
+  });
+
+  return { queues: values(queues), queries, otherTasks, counters };
+}
+
+function QueuesTable({ loading, queues }) {
+  const columns = [
+    {
+      title: 'Name',
+      dataIndex: 'name',
+    },
+    {
+      title: 'Active',
+      dataIndex: 'active',
+    },
+    {
+      title: 'Reserved',
+      dataIndex: 'reserved',
+    },
+    {
+      title: 'Waiting',
+      dataIndex: 'waiting',
+    },
+  ];
+  return <Table columns={columns} rowKey="name" dataSource={queues} loading={loading} />;
+}
+
+QueuesTable.propTypes = {
+  loading: PropTypes.bool.isRequired,
+  queues: PropTypes.arrayOf(PropTypes.any).isRequired,
+};
+
+function CounterCard({ title, value, loading }) {
+  return (
+    <Spin spinning={loading}>
+      <Card>
+        {title}
+        <div className="f-20">{value}</div>
+      </Card>
+    </Spin>
+  );
+}
+
+CounterCard.propTypes = {
+  title: PropTypes.string.isRequired,
+  value: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
+  loading: PropTypes.bool.isRequired,
+};
+
+CounterCard.defaultProps = {
+  value: '',
+};
+
+export default class AdminCeleryStatus extends React.Component {
+  state = {
+    loading: true,
+    counters: {},
+    queries: [],
+    otherTasks: [],
+    queues: [],
+  };
+
+  constructor(props) {
+    super(props);
+    this.fetch();
+  }
+
+  fetch() {
+    // TODO: handle error
+    $http.get('/api/admin/queries/tasks').then(({ data }) => {
+      const { queues, queries, otherTasks, counters } = parseTasks(data.tasks);
+      this.setState({ loading: false, queries, otherTasks, queues, counters });
+    });
+  }
+
+  render() {
+    const commonColumns = [
+      {
+        title: 'Worker Name',
+        dataIndex: 'worker',
+      },
+      {
+        title: 'PID',
+        dataIndex: 'worker_pid',
+      },
+      {
+        title: 'Queue',
+        dataIndex: 'queue',
+      },
+      {
+        title: 'State',
+        dataIndex: 'state',
+        render: (value) => {
+          if (value === 'active') {
+            return (
+              <span>
+                <Badge status="processing" /> Active
+              </span>
+            );
+          }
+          return (
+            <span>
+              <Badge status="warning" /> {value}
+            </span>
+          );
+        },
+      },
+      Columns.timeAgo({ title: 'Start Time', dataIndex: 'start_time' }),
+    ];
+
+    const queryColumns = commonColumns.concat([
+      Columns.timeAgo({ title: 'Enqueue Time', dataIndex: 'enqueue_time' }),
+      {
+        title: 'Query ID',
```

Comment:
You can already guess what I want to suggest here 😝

Reply:
Not trivial because then I need to add a function that will convert `query_id` to `Query ID`. We have something along this lines in code, but it will do `Query Id`.

Reply:
Oh, I was thinking the other way around ('Query ID' -> to lower -> snake case)